pkg/standalone-utils/test/foundry/PauseHelper.t.sol

```diff
@@ -0,0 +1,134 @@
+import { PoolFactoryMock } from "@balancer-labs/v3-vault/contracts/test/PoolFactoryMock.sol";
+import { BaseVaultTest } from "@balancer-labs/v3-vault/test/foundry/utils/BaseVaultTest.sol";
+import { IAuthentication } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/helpers/IAuthentication.sol";
+
+import { PauseHelper } from "../../contracts/PauseHelper.sol";
+
+contract PauseHelperTest is BaseVaultTest {
+    PauseHelper pauseHelper;
+
+    function setUp() public virtual override {
+        BaseVaultTest.setUp();
+
+        address[] memory owners = new address[](1);
+        owners[0] = address(this);
+
+        pauseHelper = new PauseHelper(vault);
+
+        authorizer.grantRole(pauseHelper.getActionId(pauseHelper.addPools.selector), address(this));
+        authorizer.grantRole(pauseHelper.getActionId(pauseHelper.removePools.selector), address(this));
+        authorizer.grantRole(pauseHelper.getActionId(pauseHelper.pause.selector), address(this));
+
+        authorizer.grantRole(vault.getActionId(vault.pausePool.selector), address(pauseHelper));
+    }
+
+    function testAddPoolsWithTwoBatches() public {
+        address[] memory firstPools = _addPools(10);
+        assertEq(pauseHelper.getPoolsCount(), firstPools.length, "Pools count should be 10");
+
+        address[] memory secondPools = _addPools(10);
+        assertEq(pauseHelper.getPoolsCount(), firstPools.length + secondPools.length, "Pools count should be 20");
+
+        for (uint256 i = 0; i < firstPools.length; i++) {
+            assertTrue(pauseHelper.hasPool(firstPools[i]));
+        }
+
+        for (uint256 i = 0; i < secondPools.length; i++) {
+            assertTrue(pauseHelper.hasPool(secondPools[i]));
+        }
+    }
+
+    function testAddPoolWithoutPermission() public {
+        authorizer.revokeRole(pauseHelper.getActionId(pauseHelper.addPools.selector), address(this));
+
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        pauseHelper.addPools(new address[](0));
+    }
+
+    function testRemovePools() public {
+        address[] memory pools = _addPools(10);
+        assertEq(pauseHelper.getPoolsCount(), 10, "Pools count should be 10");
+
+        for (uint256 i = 0; i < pools.length; i++) {
+            vm.expectEmit();
+            emit PauseHelper.PoolRemoved(pools[i]);
+        }
+
+        pauseHelper.removePools(pools);
+
+        assertEq(pauseHelper.getPoolsCount(), 0, "Pools count should be 0");
+
+        for (uint256 i = 0; i < pools.length; i++) {
+            assertFalse(pauseHelper.hasPool(pools[i]));
+        }
+    }
+
+    function testRemovePoolWithoutPermission() public {
+        address[] memory pools = _addPools(10);
+
+        authorizer.revokeRole(pauseHelper.getActionId(pauseHelper.removePools.selector), address(this));
+
+        vm.expectRevert(IAuthentication.SenderNotAllowed.selector);
+        pauseHelper.removePools(pools);
+    }
+
+    function testPause() public {
```

Comment:
Can also add `testDuplicatePause`. If it's already paused and you try to pause it again, the Vault will revert.

```
address[] memory pausedPools = new address[](1);
pausedPools[0] = pools[2];

vm.expectRevert(abi.encodeWithSelector(IVaultErrors.PoolPaused.selector, pausedPools[0]));
pauseHelper.pause(pausedPools);
```